src/main/java/com/example/ajaxonboardingproject/nats/NatsMovieSessionAddController.kt

```diff
@@ -23,9 +23,11 @@ class NatsMovieSessionAddController(
         MovieSessionAddRequest.parser()
 
     override fun generateReplyForNatsRequest(
-        request: MovieSessionAddRequest
+        request: Mono<MovieSessionAddRequest>
     ): Mono<MovieSessionResponse> {
-        return service.add(converter.protoToMovieSession(request.movieSession))
+        return request
+            .map { converter.protoToMovieSession(it.movieSession) }
+            .flatMap { service.add(it) }
             .map { converter.movieSessionToProtoResponse(it) }
```

Comment:
maybe method references? 
```
           .flatMap(service::add)
           .map(converter::movieSessionToProtoResponse)
```
Looks cool and sporty))

Reply:
As for me there is no difference between the two approaches. 

Somebody likes more function/method reference, somebody else likes more lambda. It's  personal preferences. So, I don't recommend arguing what is better.

Reply:
thanks for advice)